File: DailyRPG/Models/Item.cs
using System.ComponentModel.DataAnnotations;

namespace DailyRpg.Models
{
    public enum ItemType
    {
        Consumable,
        Equipament,
        Material,
        Xp

    }

    public enum EquipmentType
    {
        None,
        Weapon,
        Armor,
        Material
    }

    public class Item
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        [Required]
        public ItemType Type { get; set; }

        public EquipmentType EquipType { get; set; } = EquipmentType.None;

        public int EffectValue { get; set; } = 0;

        [Required]
        public int ShopPrice { get; set; }

        public string SkillAffinity { get; set; } = "None";

    }
}

File: frontend/lib/models/item.dart
import 'enums.dart';

class Item {

  final int id;
  final String name;
  final String description; 
  final int effectValue;
  final int shopPrice;
  final String skillAffinity; 

  final ItemType type;
  final EquipmentType equipType;

  Item({
    required this.id,
    required this.name,
    required this.description,
    required this.type,
    required this.equipType,
    required this.effectValue,
    required this.shopPrice,
    required this.skillAffinity,
  });

  factory Item.fromJson(Map<String, dynamic> json) {

      return Item(

        id: json['id'],

        name: json['name'],

        description: json['description'] ?? '',
        
        type: itemTypeFromInt(json['type']),
        
        equipType: equipTypeFromInt(json['equipType']),

        effectValue: json['effectValue'],
        shopPrice: json['shopPrice'],
        skillAffinity: json['skillAffinity'] ?? 'None',
      );
    }
}